### DISCUSSION TITLE
how to keep meta information in custom transform

### DISCUSSION THREAD

**Discussion by JunMa11** [Project-MONAI/MONAI#5299] [Q&A]
Dear MONAI Team,

Thanks for the awesome work!

I created a simple transform followed this tutorial
https://github.com/Project-MONAI/MONAIBootcamp2021/blob/main/day1/1.%20Getting%20Started%20with%20MONAI.ipynb

```python
test_files = [join(input_path, i)for i in img_names]

def preprocess(img):
    # img is an 2D RGB image (H, W, 3)
    # preprocessing: downsample and convert to gray image
    return np.expand_dims(color.rgb2gray(img[::4,::4, :]), 0).astype(np.float32)

test_transforms = Compose(
    [
        LoadImage(image_only=True, dtype=np.uint8),
        Lambda(preprocess),
        ToTensor()
    ]
)

test_tensor = test_transforms(test_files)
```
However, I found that part of the meta information were missed in `test_tensor`, e.g., `filename_or_obj`

1. How to keep the meta information in the custom transform? 

2. Furthermore, after obtaining the segmentations of the batch input, is there a way to save the segmentation results in a batch? The expected segmentation names are the same as the image names.

```python
test_ds = monai.data.Dataset(data=test_files, transform=test_transforms)
test_loader = DataLoader(test_ds, batch_size=4, shuffle=False, num_workers=4)
check_data = monai.utils.misc.first(test_loader)
post_pred = Compose([EnsureType(), Activations(softmax=True), AsDiscrete(threshold=0.5)])
model.eval()
with torch.no_grad():
    for test_tensor in test_loader:
        test_pred_out = sliding_window_inference(test_tensor.to(device), (args.input_size, args.input_size), sw_batch_size, model) # (B,C,H,W)
        test_pred_out = [post_pred(i) for i in monai.data.decollate_batch(test_pred_out)] # batch to list
        # how to save test_pred_out in a batch mode, the expected names are the same as the input names
```

**Comment by Nic-Ma** [comment]
Hi @JunMa11 ,

You can follow the new `MetaTensor` data type to implement your transform and keep the meta information:
https://github.com/Project-MONAI/tutorials/blob/main/modules/transforms_metatensor.ipynb
https://github.com/Project-MONAI/tutorials/blob/main/modules/transforms_update_meta_data.ipynb
CC @wyli @rijobro 

Thanks.
